starboard/export.h

```diff
@@ -62,12 +62,10 @@
 
 // Specification for an Android-specific symbol that should be exported for
 // use at layers above Starboard. Only needed for component builds.
-#if BUILDFLAG(IS_ANDROID)
-#if defined(COMPONENT_BUILD)
+#if BUILDFLAG(IS_ANDROID) && defined(COMPONENT_BUILD)
```

Comment:
@y4vor or @kaidokert any thoughts on exporting non-android symbols for Android component builds?
We could also say that android debug builds also aren't component builds (as with linux), but I think we'll run into issues when we eventually try to upstream and aren't properly exporting symbols from components, so I'm hesitant to do that.